common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -447,6 +448,17 @@ public static Geometry lineFromMultiPoint(Geometry geometry) {
         return GEOMETRY_FACTORY.createLineString(coordinates.toArray(new Coordinate[0]));
     }
 
+    public static Geometry closestPoint(Geometry left, Geometry right) {
+        DistanceOp distanceOp = new DistanceOp(left, right);
+        try {
+            Coordinate[] closestPoints = distanceOp.nearestPoints();
+            return GEOMETRY_FACTORY.createPoint(closestPoints[0]);
+        }
+        catch (Exception e) {
+            return null;
```

Comment:
Do not throw null here, throw a meaningful exception. If this is thrown if the geometry is empty, throw an IllegalArgumentException with an appropriate message

Reply:
Should we throw an exception or just return a null value? I think postgis return an empty value:
```
postgres=# select  ST_AsText(ST_ClosestPoint(ST_GeomFromText('POINT EMPTY'), 
                                   ST_GeomFromText('LINESTRING (0 0, 1 0, 2 0, 3 0, 4 0, 5 0)')));
 st_astext 
-----------
 
(1 row)
```

Reply:
It is better to throw an exception, in a case where multiple functions are stringed together, which is what happens usually, throwing null means user would see an unexplainable NullPointerException eventually.

By throwing an IllegalArgumentException with a proper message, we will end up helping the user.

Reply:
Good idea!

Reply:
Based on this, I tried to resolve all your comments! You can review again.  :) Thanks! 😁